pkg/client/leaderelection/leaderelection.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package leaderelection implements leader election of a set of endpoints.
+// It uses an annotation in the endpoints object to store the record of the
+// election state.
+package leaderelection
```

Comment:
> applying no algorithm of election

I'd consider this to be a variant of [randomized leader election](https://en.wikipedia.org/wiki/Leader_election#Randomized_.28probabilistic.29_leader_election) where the identity is the order that the processes make "aquire" requests to etcd. Randomness is added by processor speed and network latency. We pick the first.

I think this satisfies the definition of leader election and I want to call it that because that's what people know it by.

> If you can decouple the mechanics and leverage through an interface

Agree that we should do this. The should last longer than the implementation. I'll see what i can do